fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AstToStringBuilder.java

```diff
@@ -292,41 +310,227 @@ public String visitExportStatement(ExportStmt stmt, Void context) {
 
         @Override
         public String visitGrantRevokePrivilegeStatement(BaseGrantRevokePrivilegeStmt stmt, Void context) {
-            StringBuilder sb = new StringBuilder();
-            if (stmt instanceof GrantPrivilegeStmt) {
-                sb.append("GRANT ");
+            if (GlobalStateMgr.getCurrentState().isUsingNewPrivilege()) {
+                StringBuilder sb = new StringBuilder();
+                if (stmt instanceof GrantPrivilegeStmt) {
+                    sb.append("GRANT ");
+                } else {
+                    sb.append("REVOKE ");
+                }
+
+                List<String> privList = new ArrayList<>();
+                for (Map.Entry<String, Action> actionEntry : stmt.getObjectType().getActionMap().entrySet()) {
+                    if (stmt.getActionList().contains(actionEntry.getValue())) {
+                        privList.add(actionEntry.getValue().getName());
+                    }
+                }
+                sb.append(Joiner.on(", ").join(privList));
+                sb.append(" ON ");
+
+                switch (stmt.getObjectType()) {
+                    case TABLE:
+                    case VIEW:
+                    case MATERIALIZED_VIEW: {
+                        TablePEntryObject tablePEntryObject = (TablePEntryObject) stmt.getObjectList().get(0);
```

Comment:
why only handle the first one, what if the sql is : grant select on testaa.testaa, testaa.testbb to testuser3;? can we should it correctly